lib/recursive-methods.rb

```diff
@@ -36,14 +65,24 @@ def search(array, value)
     raise NotImplementedError, "Method not implemented"
 end
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) where n is the size of the input
+# Space complexity: O(1) because additional space in memory is not being consumed
```

Comment:
O(n^2) for both time and space since you use the system stack and create a new string with each iteration.